manager/scheduler/constraint.go

```diff
@@ -44,26 +44,36 @@ func (f *ConstraintFilter) Check(n *NodeInfo) bool {
 			if !constraint.Match(n.Description.Hostname) {
 				return false
 			}
+		case "node.role":
```

Comment:
Ideally they would be rescheduled somewhere else. Why don't we re-kick the scheduler on label updates? At least on labels inside of `Specs`...